WordPress/Classes/ViewRelated/Stats/JetpackLoginViewController.swift

```diff
@@ -0,0 +1,546 @@
+import Foundation
+import UIKit
+import SVProgressHUD
+import WordPressShared
+import WordPressComAnalytics
+
+/// A view controller that presents a Jetpack login form
+///
+class JetpackLoginViewController: UIViewController {
+
+    // MARK: - Constants
+
+    fileprivate let jetpackInstallRelativePath = "plugin-install.php?tab=plugin-information&plugin=jetpack"
+    fileprivate let jetpackMoreInformationURL = "https://apps.wordpress.com/support/#faq-ios-15"
+
+    // MARK: - Properties
+
+    typealias CompletionBlock = (Bool) -> Void
+    /// This completion handler closure is executed when the authentication process handled
+    /// by this VC is completed.
+    ///
+    open var completionBlock: CompletionBlock?
+
+    @IBOutlet fileprivate weak var jetpackImage: UIImageView!
+    @IBOutlet fileprivate weak var descriptionLabel: UILabel!
+    @IBOutlet fileprivate weak var usernameTextField: WPWalkthroughTextField!
+    @IBOutlet fileprivate weak var passwordTextField: WPWalkthroughTextField!
+    @IBOutlet fileprivate weak var verificationCodeTextField: WPWalkthroughTextField!
+    @IBOutlet fileprivate weak var scrollView: UIScrollView!
+    @IBOutlet fileprivate weak var signinButton: WPNUXMainButton!
+    @IBOutlet fileprivate weak var sendSMSCodeButton: UIButton!
+    @IBOutlet fileprivate weak var installJetpackButton: WPNUXMainButton!
+    @IBOutlet fileprivate weak var moreInformationButton: UIButton!
+
+    fileprivate var blog: Blog!
+    fileprivate var activeField: UITextField?
+    fileprivate var shouldDisplayMultifactor: Bool = false
+    fileprivate var loginFields = LoginFields()
+
+    fileprivate lazy var loginFacade: LoginFacade = {
+        let facade = LoginFacade()
+        facade.delegate = self
+        return facade
+    }()
+
+    /// Returns true if this VC is currently authenticating with the server.
+    /// After setting, the UI controls will be updated accordingly
+    ///
+    fileprivate var isAuthenticating: Bool = false {
+        didSet {
+            self.usernameTextField.isEnabled = !self.isAuthenticating
+            self.passwordTextField.isEnabled = !self.isAuthenticating
+            self.verificationCodeTextField.isEnabled = !self.isAuthenticating
+            self.signinButton.showActivityIndicator(self.isAuthenticating)
+        }
+    }
+
+    /// Returns true if the blog has the proper version of Jetpack installed,
+    /// otherwise false
+    ///
+    fileprivate var hasJetpack: Bool {
+        guard let jetpack = self.blog.jetpack else {
+            return false
+        }
+        return (jetpack.isInstalled() && jetpack.isUpdatedToRequiredVersion())
+    }
+
+    // MARK: - Initializers
+
+    /// Preferred initializer for JetpackLoginViewController
+    ///
+    /// - Parameter blog: The current blog
+    ///
+    convenience init(blog: Blog) {
```

Comment:
Do you want this to be a convenience initializer or really just make it the only way to init the VC? Seems like we're trying to imply a Blog object is required and shouldn't be used otherwise.